![logo-gt.png](attachment:logo-gt.png)

<div class="alert alert-block alert-info">
    <h1>BIOS 4150/BIOL 6150</h1>
    <h3>Instructor: Dr. King Jordan</h3>
    <p>TAs: Shivam Sharma (shivamsharma13@gatech.edu) & Vishva Natarajan (vishvanata@gatech.edu)</p>
</div>

<div class="alert alert-block alert-warning">
    <h2>Project 1 (Data access & QC) starter notebook</h2>
    <h3>Deadline: 12:00AM, October 20th, 2023</h3>
</div>

---

# **1. Your assigned 1000 Genomes individual**
### *Total Questions: 3*
### *Total Points: 10+10+20 = 40*


---

<div class="alert alert-block alert-warning">
    <h3>1.1 Does you group have an assigned individual ID from the 1000 Genomes Project?</h3>
    <p>Find the individual on the international genomes website and paste a screenshot below</p>
</div>

![HGPindividual.png](attachment:HGPindividual.png)


<br>If screenshot is not visible, the images are also uploaded in the shared directory under group18 folder <br>


<div class="alert alert-block alert-warning">
    <h3>1.2 What population does you assigned individual belong to?</h3>
    <p>Write about where the geographical location of where this individual lived at the time of recruitment? </p>
</div>

#Write about the individual and their ancestors here.

Sex - Male <br> Population - Colombian in Medellin, Colombia, American Ancestry

<div class="alert alert-block alert-warning">
    <h3>1.3 More about your individual's population?</h3>
    <p>Read more about this population and write about their recent historical  (100-500 years) ancestors.</p>
</div>

#Write about the individual and their ancestors here.

American Ancestry - American ancestry refers to people in the United States who self-identify their ancestral origin or descent as "American", rather than the more common officially recognized racial and ethnic groups that make up the bulk of the American people. The majority of these respondents are visibly White Americans, who are far removed from and no longer self-identify with their original ethnic ancestral origins. (Sourced from Wikipedia)

# **2. Exome sequencing**
### *Total Questions: 3*
### *Total Points: 20+20= 40*


---

<div class="alert alert-block alert-warning">
    <h3>2.1 Does your assigned individual has exome sequencing data? If yes, then what is the SRR ID for that sequencing run?</h3>
    <p>Does you individual have multiple SRR IDs? In case they do, which one would you pick? (Hint: The sequencing information is coming from the same individual; but larger file size might be more preferable)</p>
</div>

<div class="alert alert-block alert-danger">
    <h3>Multiple fastq files for your 1KGP individual</h3>
    <p>On the FTP server, it is expected that you will have more than 1 pair of fastq files. What you need to decide on is which set of fastq files make the most sense. A few good checks are the size of the file (maybe 1-12GB), presence of an SRR ID, elimination of exceptionally small files etc. </p>
    <p>If your team is still confused, please reach out to us regarding this.</p>
</div>

#Write the SRR ID here along with your reasoning for selection.

1000Genome Phase 3 release for individual HG01112 had 2 different SRR IDs -
1. SRR070501
2. SRR070794

Both of them had both single-end and paired-end reads and approximately the same size. Initially, we picked SRR070794 which had slightly larger file size for paired-end reads(4.05GB) compared to SRR070501(4.00GB). However, after running the QC for both the 
SRRs we found SRR070501 had better quality reads and hence we went ahead with SRR070501.

<div class="alert alert-block alert-warning">
    <h3>2.2 Looking at just the exome sequencing of this individual, how many variants would you expect them to carry compared to the reference genome?</h3>
    <p>Keep in mind this is 1 individual, and learn about the expected number of variants from WGS, WES, and Genotyping</p>
</div>

#Write about the variant information here. <br>
<br> Even though there is a lot of variability in the number of variants based on the read depth, quality of reads, a typical WES dataset contains around 100k to 130k to  variants for an individual. However, we will need to perform variant calling in order to get an idea of the exact number of variants.

# **3. Data access**
### *Total Questions: 3*
### *Total Points: 20+20 = 40*


---

<div class="alert alert-block alert-warning">
    <h3>3.1 Download the exome sequencing data for this individual and write the size of the files?</h3>
    <p>3.1.1 How many files do you have? Is the sequencing paired?</p>
    <p>3.1.2 How many reads are present in each file? </p>
</div>

In [ ]:
#Show you download commamd.
!wget ftp://ftp.1000genomes.ebi.ac.uk/vol1/ftp/phase3/data/HG01112/sequence_read/SRR070501_1.filt.fastq.gz
!wget ftp://ftp.1000genomes.ebi.ac.uk/vol1/ftp/phase3/data/HG01112/sequence_read/SRR070501_2.filt.fastq.gz

In [ ]:
!mkdir ~/scratch/project1

In [2]:
#Move the files to an appropriate place and 'ls' that directory. We want to see the file names and the file sizes
!mv SRR070501_1.filt.fastq.gz ~/scratch/project1
!mv SRR070501_2.filt.fastq.gz ~/scratch/project1

In [ ]:
!ls -l ~/scratch/project1

# We also ran md5sum command to verify if the downloaded file was complete and compared it with the checksum provided on 1000genome website
# SRR070501_1.filt.fastq.gz - d95922be89b82ee36a40b9654d52ed1b
# SRR070501_2.filt.fastq.gz - ee67144c72a8d628ea71368188548c54

#!md5sum ~/scratch/project1/SRR070501_1.filt.fastq.gz
#d95922be89b82ee36a40b9654d52ed1b  SRR070501_1.filt.fastq.gz

#!md5sum ~/scratch/project1/SRR070501_2.filt.fastq.gz
#ee67144c72a8d628ea71368188548c54  SRR070501_2.filt.fastq.gz

In [ ]:
#Write about the fastq file here.
!zcat ~/scratch/project1/SRR070501_1.filt.fastq.gz | head
!echo "--------------------------------------------------------------"
!zcat ~/scratch/project1/SRR070501_2.filt.fastq.gz | head

In [ ]:
#Write about the fastq file here.
# The fastq file will contain -
# 1. A sequence identifier with information about the sequencing run.
# 2. The sequence(the base calls; A, C, T, G and N).
# 3. A separator, which is simply a plus (+) sign.
# 4. The base call . These are Phred +33 encoded, using ASCII characters to represent the numerical quality scores.
# 5. The total reads present in forward read: 23722412 ,reverse read: 23722412 
!zcat ~/scratch/project1/SRR070501_1.filt.fastq.gz | grep "^@SRR" | head
!zcat ~/scratch/project1/SRR070501_2.filt.fastq.gz | grep "^@SRR" | head

# **4. Pre-QC with FastQC**
### *Total Questions: 2*
### *Total Points: 20+20+20 = 60*


---

<div class="alert alert-block alert-warning">
    <h3>4.1 Show your exact command below.</h3>
</div>

In [1]:
#Command here.
!mkdir -p ~/scratch/project1/fastqc_before_trimming
!mkdir -p ~/scratch/project1/trimming
!mkdir -p ~/scratch/project1/fastqc_after_trimming

!fastqc -o ~/scratch/project1/fastqc_before_trimming ~/scratch/project1/SRR070501_1.filt.fastq.gz ~/scratch/project1/SRR070501_2.filt.fastq.gz

application/gzip
application/gzip
Started analysis of SRR070501_1.filt.fastq.gz
Approx 5% complete for SRR070501_1.filt.fastq.gz
Approx 10% complete for SRR070501_1.filt.fastq.gz
Approx 15% complete for SRR070501_1.filt.fastq.gz
Approx 20% complete for SRR070501_1.filt.fastq.gz
Approx 25% complete for SRR070501_1.filt.fastq.gz
Approx 30% complete for SRR070501_1.filt.fastq.gz
Approx 35% complete for SRR070501_1.filt.fastq.gz
Approx 40% complete for SRR070501_1.filt.fastq.gz
Approx 45% complete for SRR070501_1.filt.fastq.gz
Approx 50% complete for SRR070501_1.filt.fastq.gz
Approx 55% complete for SRR070501_1.filt.fastq.gz
Approx 60% complete for SRR070501_1.filt.fastq.gz
Approx 65% complete for SRR070501_1.filt.fastq.gz
Approx 70% complete for SRR070501_1.filt.fastq.gz
Approx 75% complete for SRR070501_1.filt.fastq.gz
Approx 80% complete for SRR070501_1.filt.fastq.gz
Approx 85% complete for SRR070501_1.filt.fastq.gz
Approx 90% complete for SRR070501_1.filt.fastq.gz
Approx 95% complete f

<div class="alert alert-block alert-warning">
    <h3>4.2 Show <i>per base sequencing quality</i>?</h3>
    <p>Add a screenshot below</p>
</div>

per base sequencing quality for forward read before qc

![PBSQ_beforeQC.png](attachment:PBSQ_beforeQC.png)

per base sequence quality for reverse read before qc
![PBSQ_beforeQC2.png](attachment:PBSQ_beforeQC2.png)

<div class="alert alert-block alert-warning">
    <h3>4.3 Show <i>Per base sequence content</i>?</h3>
    <p>Add a screenshot below</p>
</div>

per base sequence content for forward read before qc
![PBSC_beforeQC.png](attachment:PBSC_beforeQC.png)

per base sequence content for reverse read before qc
![PBSC_beforeQC2.png](attachment:PBSC_beforeQC2.png)

# **5. Trimming with fastp or trimmmomatic**
### *Total Questions: 3*
### *Total Points: 20+20+20 = 60*


---

<div class="alert alert-block alert-warning">
    <h3>5.1 What was the tools and command you used?</h3>
    <p>Write the tool name and exact command below</p>
</div>

We used fastp for trimming

In [2]:
#Exact command here
!fastp \
-i ~/scratch/project1/SRR070501_1.filt.fastq.gz \
-I ~/scratch/project1/SRR070501_2.filt.fastq.gz \
-o ~/scratch/project1/trimming/SRR070501_1_trimmed.fastq.gz \
-O ~/scratch/project1/trimming/SRR070501_2_trimmed.fastq.gz \
-f 13 \
-t 10

Read1 before filtering:
total reads: 23722412
total bases: 2372241200
Q20 bases: 2140979447(90.2513%)
Q30 bases: 1979960362(83.4637%)

Read2 before filtering:
total reads: 23722412
total bases: 2372241200
Q20 bases: 2147496408(90.5261%)
Q30 bases: 1975524098(83.2767%)

Read1 after filtering:
total reads: 21179531
total bases: 1630140066
Q20 bases: 1562596318(95.8566%)
Q30 bases: 1462407501(89.7105%)

Read2 after filtering:
total reads: 21179531
total bases: 1630140066
Q20 bases: 1562595186(95.8565%)
Q30 bases: 1451537439(89.0437%)

Filtering result:
reads passed filter: 42359062
reads failed due to low quality: 5081638
reads failed due to too many N: 4124
reads failed due to too short: 0
reads with adapter trimmed: 189186
bases trimmed due to adapters: 1484716

Duplication rate: 1.23928%

Insert size peak (evaluated by paired-end reads): 149

JSON report: fastp.json
HTML report: fastp.html

fastp -i /home/hice1/akrishna311/scratch/project1_test/SRR070501_1.filt.fastq.gz -I /home/hice1/

<div class="alert alert-block alert-warning">
    <h3>5.2 How did you perform the trimming?</h3>
    <p>Did you use dynamic built in features or hard coded parameters?</p>
    <p>Did you have to go back and repeat the trimming after run(s) for <b>postQC?</b></p>
</div>


We checked the QC results after running fastqc and decided the reads to be trimmed and hardcoded the values (13 bases from the beginning and 10 from the end). We did not have to revisit trimming step as the results were good after trimming the first time.


<div class="alert alert-block alert-warning">
    <h3>5.3 What percentage of reads did you lose in the trimming process?</h3>
    <p>5.3.1 Write the total number of reads before and after trimming</p>
    <p>5.3.2 If you had a paired fastq file, are there any reads after trimming present in one file and not the other? (eg: present in reverse but not in forward?)</p>
    
</div>

#Before trimming:
Read1 total reads: 23722412
Read2 total reads: 23722412

#After trimming:
Read1 total reads: 21179531
Read2 total reads: 21179531

Percentage of reads lost after trimming for both reads = (before trimming - after trimming)/before trimming * 100 = 10.72%

In [3]:
#We are grepping for header lines until the ""/{number}"" to compare the reads present in both the files. 
!zcat ~/scratch/project1/trimming/SRR070501_1_trimmed.fastq.gz | grep -Eo "^@SRR.*\/" > ~/scratch/project1/trimming/headers1.txt
!zcat ~/scratch/project1/trimming/SRR070501_2_trimmed.fastq.gz | grep -Eo "^@SRR.*\/" > ~/scratch/project1/trimming/headers2.txt

#No output from the below command means that there are no reads that are unique to either files
!comm -3 <(sort ~/scratch/project1/trimming/headers1.txt) <(sort ~/scratch/project1/trimming/headers2.txt) 

# **6. Post-QC with FastQC?**
### *Total Questions: 2*
### *Total Points: 20+20+20 = 60*


---

<div class="alert alert-block alert-warning">
    <h3>6.1 Show your exact command below.</h3>
</div>

In [4]:
#command here
!fastqc -o ~/scratch/project1/fastqc_after_trimming ~/scratch/project1/trimming/SRR070501_1_trimmed.fastq.gz ~/scratch/project1/trimming/SRR070501_2_trimmed.fastq.gz

application/gzip
application/gzip
Started analysis of SRR070501_1_trimmed.fastq.gz
Approx 5% complete for SRR070501_1_trimmed.fastq.gz
Approx 10% complete for SRR070501_1_trimmed.fastq.gz
Approx 15% complete for SRR070501_1_trimmed.fastq.gz
Approx 20% complete for SRR070501_1_trimmed.fastq.gz
Approx 25% complete for SRR070501_1_trimmed.fastq.gz
Approx 30% complete for SRR070501_1_trimmed.fastq.gz
Approx 35% complete for SRR070501_1_trimmed.fastq.gz
Approx 40% complete for SRR070501_1_trimmed.fastq.gz
Approx 45% complete for SRR070501_1_trimmed.fastq.gz
Approx 50% complete for SRR070501_1_trimmed.fastq.gz
Approx 55% complete for SRR070501_1_trimmed.fastq.gz
Approx 60% complete for SRR070501_1_trimmed.fastq.gz
Approx 65% complete for SRR070501_1_trimmed.fastq.gz
Approx 70% complete for SRR070501_1_trimmed.fastq.gz
Approx 75% complete for SRR070501_1_trimmed.fastq.gz
Approx 80% complete for SRR070501_1_trimmed.fastq.gz
Approx 85% complete for SRR070501_1_trimmed.fastq.gz
Approx 90% comple

<div class="alert alert-block alert-warning">
    <h3>6.2 Show <i>per base sequencing quality</h3>
    <p>If the quality was falling below 20 for certain regions then trim it. Add a screenshot below</p>
</div>

per base sequencing quality for forward read after qc
![PBSQ_afterQC.png](attachment:PBSQ_afterQC.png)

per base sequencing quality for reverse read after qc
![PBSQ_afterQC2.png](attachment:PBSQ_afterQC2.png)

<div class="alert alert-block alert-warning">
    <h3>6.3 Show <i>Per base sequence content</i>?</h3>
    <p>Ideally we shouldn't see adapters at the beginning of the reads in the fastq files. Add a screenshot below</p>
</div>

per base sequence content for forward read after qc 
![PBSC_afterQC.png](attachment:PBSC_afterQC.png)

per base sequence content for reverse read after qc
![PBSC_afterQC2.png](attachment:PBSC_afterQC2.png)